genai-on-vertex-ai/vertex-ai-search/Ingesting_unstructured_docuemtns_with_metadata.ipynb

```diff
@@ -0,0 +1,1054 @@
+{
```

Comment:
<p>Since you are using REST API throughout, I suggest using <code>AuthorizedSession</code> instead of  <code>creds.refresh </code>  + <code>requests.post</code> .  </p><p> </p>

<pre class="ql-syntax" spellcheck="false">from google.auth import default
from google.auth.transport.requests import AuthorizedSession

creds, _ = default()
authed_session = AuthorizedSession(creds)

query_url = f'https://discoveryengine.googleapis.com/v1/projects/{project_id}/locations/{location}/collections/default_collection/dataStores/{datastore_id}/servingConfigs/default_search:search'
http_headers = {"Content-Type": "application/json"}
data = {"query": query,
&nbsp; &nbsp; &nbsp; &nbsp; "pageSize": 1}
response = authed_session.post(query_url, headers=http_headers, json=data)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/67/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Reply:
<p>Great, thanks</p>